ddev/tests/cli/validate/test_metrics.py

```diff
@@ -628,3 +691,111 @@ def test_metrics_passing(ddev, helpers):
         Passed: 1
         """
     )
+
+
+def test_metrics_ordered(fake_repo, ddev, helpers):
+    write_file(
+        fake_repo.path / "metadata_integration",
+        'metadata.csv',
+        """metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name,curated_metric
+metadata_integration.metric_a,gauge,,,,My metric A,0,metadata_integration,,
+metadata_integration.metric_b,gauge,,,,My metric B,0,metadata_integration,,
+""",
+    )
+
+    result = ddev('validate', 'metadata', 'metadata_integration')
+
+    assert result.exit_code == 0, result.output
+    assert helpers.remove_trailing_spaces(result.output) == helpers.dedent(
+        """
+        Metrics validation
+
+        Passed: 1
+        """
+    )
+
+
+def test_metrics_not_ordered(fake_repo, ddev, helpers):
+    result = ddev('validate', 'metadata', 'metadata_integration')
+
+    assert result.exit_code == 1, result.output
+    assert helpers.remove_trailing_spaces(result.output) == helpers.dedent(
+        """
+        Metrics validation
+        └── metadata_integration
+            └── metadata_integration/metadata.csv
+
+                metadata_integration: metadata.csv is not sorted by metric name. Run
+                `ddev validate metadata metadata_integration --sync` to sort it.
+
+        Errors: 1
+        """
+    )
+
+
+def test_metrics_not_ordered_sync(fake_repo, ddev, helpers):
+    result = ddev('validate', 'metadata', 'metadata_integration', '--sync')
+
+    assert result.exit_code == 1, result.output
+    assert helpers.remove_trailing_spaces(result.output) == helpers.dedent(
+        """
+        Metrics validation
+        └── metadata_integration
+            └── metadata_integration/metadata.csv
+
+                metadata_integration: metadata.csv is not sorted by metric name. Run
+                `ddev validate metadata metadata_integration --sync` to sort it.
+                Sorting metadata_integration/metadata.csv by metric names
+
+        Errors: 1
+        """
+    )
```

Comment:
I'm a little confused:

1. Shouldn't we exit with code `0` if we **succeed** in sorting a metadata.csv file?
2. As a user I'd be very confused if the `validate metadata --sync` command told me to run the `validate metadata --sync` command...

Reply:
Addressed!

Reply:
Hmm, the version I see still has error code `1` and still says to run the command with `--sync`. It now includes a message that it's sorting, but imo that's **the only** message it needs to display.

Reply:
Link? I don't see it 😓

Reply:
ah, it was the old version indeed 🤦 

nvm, looking good!